<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>704</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="font">
   <font>
    <pointsize>-1</pointsize>
    <weight>75</weight>
    <italic>false</italic>
    <bold>true</bold>
   </font>
  </property>
  <property name="windowTitle">
   <string>Virtual Concierge Creator</string>
  </property>
  <property name="windowIcon">
   <iconset resource="content.qrc">
    <normaloff>:/Icon</normaloff>:/Icon</iconset>
  </property>
  <property name="windowOpacity">
   <double>1.000000000000000</double>
  </property>
  <property name="styleSheet">
   <string notr="true">QWidget {
  background: #3C3F41;
  color: white;
  font: bold;
font-size: 12px;
}
QLineEdit{
  background: #2D2E30;
background-color: #2D2E30;
border-radius: 4px;
}
QLineEdit:focus{
  background: #0073C7; 
  background-color: #0073C7;
  border-radius: 4px;
}
QTextBrowser{
  background: #2D2E30;
background-color: #2D2E30;
border-radius: 4px;
}
/* Button Styles*/
QPushButton {
  border: 1px solid #2D2E30;
  background: #2D2E30;
  color: white;
  font: bold;
    border-radius: 4px;
}
QPushButton:hover:!pressed { 
  border: 2px solid #0073C7;
  background: #0073C7;
    border-radius: 4px;
}
QPushButton:checked { 
  border: 2px solid #0073C7;
  background: #0073C7;
    border-radius: 4px;
}
/* Menu Bar Styles */
QMenuBar {
    background-color:#3C3F41;
}

QMenuBar::item {
    spacing: 3px; /* spacing between menu bar items */
    padding: 10px 32px 10px 32px;
    background: transparent;
    border-radius: 4px;
    color: white;
}

QMenuBar::item:selected { /* when selected using mouse or keyboard */
    background: #0073C7;
}
QMenuBar::item:hover:!pressed{
  border: 2px solid #0073C7;
  background: #0073C7;
}

QMenu::item { 
  padding: 10px 32px 2px 20px;
  font: 14px;
 }
QMenu::item:selected { 
  background: #0073C7; 
  border-radius: 4px;
  color: white;
}
/* Spin Box Styles */
QAbstractSpinBox {
    background-color:#2D2E30;
    border-radius: 4px;
}
QAbstractSpinBox:focus{
  background: #0073C7;
background-color: #0073C7;
border-radius: 4px;
}
QAbstractSpinBox::up-button { 
  border: 0px solid #2D2E30;
}
QAbstractSpinBox::down-button { 
  border: 0px solid #2D2E30;
}
QAbstractSpinBox::up-arrow { 
  image:url(:/Content/spin_arrow_up_white.png);
  width:8;
  height:8;
  border-width: 0;
}
QAbstractSpinBox::down-arrow { 
  image:url(:/Content/spin_arrow_down_white.png);
  width:8;
  height:8;
  border-width: 0;
}
QAbstractSpinBox::up-arrow:hover:!pressed { 
  image:url(:/Content/spin_arrow_up_blue.png);
  width:8;
  height:8;
  border-width: 0;
}
QAbstractSpinBox::down-arrow:hover:!pressed { 
  image:url(:/Content/spin_arrow_down_blue.png);
  width:8;
  height:8;
  border-width: 0;
}
QComboBox {
    border: 1px solid gray;
    border-radius: 3px;
    padding: 1px 18px 1px 3px;
    min-width: 6em;
}

QComboBox:editable {
    background: white;
}

QComboBox:!editable, QComboBox::drop-down:editable {
     background: #2D2E30;
}

/* QComboBox gets the &quot;on&quot; state when the popup is open */
QComboBox:!editable:on, QComboBox::drop-down:editable:on {
    background:#0073C7;
}

QComboBox:on { /* shift the text when the popup opens */
    padding-top: 3px;
    padding-left: 4px;
}

QComboBox::drop-down {
    subcontrol-origin: padding;
    subcontrol-position: top right;
    width: 15px;

    border-left-width: 1px;
    border-left-color: darkgray;
    border-left-style: solid; /* just a single line */
    border-top-right-radius: 3px; /* same radius as the QComboBox */
    border-bottom-right-radius: 3px;
}

QComboBox::down-arrow {
    image: url(:/Content/spin_arrow_down_white.png);
  width: 8;
height: 8;
}

QComboBox::down-arrow:on { /* shift the arrow when popup is open */
    top: 1px;
    left: 1px;
}
QComboBox QAbstractItemView {
    border: 2px solid darkgray;
    selection-background-color:  #0073C7;
}</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="1">
     <widget class="RenderState" name="openGLWidget"/>
    </item>
   </layout>
  </widget>
  <widget class="QDockWidget" name="dockWidget_basic_tools">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="minimumSize">
    <size>
     <width>320</width>
     <height>360</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>320</width>
     <height>360</height>
    </size>
   </property>
   <property name="windowIcon">
    <iconset resource="content.qrc">
     <normaloff>:/Content/delete_icon_white.png</normaloff>
     <normalon>:/Content/run_icon_white.png</normalon>
     <disabledoff>:/Content/plus_icon_white.png</disabledoff>
     <disabledon>:/Content/delete_icon_white.png</disabledon>
     <activeoff>:/Content/delete_icon_white.png</activeoff>
     <activeon>:/Content/delete_icon_white.png</activeon>:/Content/delete_icon_white.png</iconset>
   </property>
   <property name="accessibleName">
    <string/>
   </property>
   <property name="windowTitle">
    <string>Add Objects</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QGridLayout" name="gridLayout_5">
     <item row="1" column="0">
      <widget class="QPushButton" name="button_add_basic">
       <property name="minimumSize">
        <size>
         <width>18</width>
         <height>24</height>
        </size>
       </property>
       <property name="text">
        <string>Add</string>
       </property>
       <property name="icon">
        <iconset resource="content.qrc">
         <normaloff>:/Content/plus_icon_white.png</normaloff>:/Content/plus_icon_white.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>16</width>
         <height>16</height>
        </size>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QPushButton" name="button_edit_basic">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>24</height>
        </size>
       </property>
       <property name="text">
        <string>Edit</string>
       </property>
       <property name="icon">
        <iconset resource="content.qrc">
         <normaloff>:/Content/edit_icon_white.png</normaloff>:/Content/edit_icon_white.png</iconset>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QPushButton" name="button_remove_basic">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>24</height>
        </size>
       </property>
       <property name="text">
        <string>Remove</string>
       </property>
       <property name="icon">
        <iconset resource="content.qrc">
         <normaloff>:/Content/delete_icon_white.png</normaloff>:/Content/delete_icon_white.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>16</width>
         <height>16</height>
        </size>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="2" column="0" colspan="3">
      <widget class="QStackedWidget" name="stackedWidget_side_add">
       <property name="midLineWidth">
        <number>0</number>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="page_add_node">
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="0">
          <widget class="QGroupBox" name="groupBox_node_settings_2">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="title">
            <string>Node Settings</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_2">
            <item row="0" column="0">
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Node Name</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1" colspan="2">
             <widget class="QDoubleSpinBox" name="doubleSpinBox_node_y">
              <property name="minimumSize">
               <size>
                <width>32</width>
                <height>32</height>
               </size>
              </property>
              <property name="minimum">
               <double>-9999.000000000000000</double>
              </property>
              <property name="maximum">
               <double>9999.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.100000000000000</double>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_node_pos_x">
              <property name="text">
               <string>X Position</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1" colspan="2">
             <widget class="QDoubleSpinBox" name="doubleSpinBox_node_x">
              <property name="minimumSize">
               <size>
                <width>32</width>
                <height>32</height>
               </size>
              </property>
              <property name="minimum">
               <double>-9999.000000000000000</double>
              </property>
              <property name="maximum">
               <double>9999.989999999999782</double>
              </property>
              <property name="singleStep">
               <double>0.100000000000000</double>
              </property>
             </widget>
            </item>
            <item row="0" column="1" colspan="2">
             <widget class="QLineEdit" name="lineEdit_node_name">
              <property name="minimumSize">
               <size>
                <width>32</width>
                <height>32</height>
               </size>
              </property>
              <property name="text">
               <string>default_node</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_node_pos_y">
              <property name="text">
               <string>Y Position</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="page_edit_link">
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="0" column="0">
          <widget class="QGroupBox" name="groupBox_edit_edge">
           <property name="title">
            <string>Edit Edge</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_6">
            <item row="0" column="0">
             <layout class="QHBoxLayout" name="horizontalLayout">
              <item>
               <widget class="QLabel" name="label_edge_weight">
                <property name="text">
                 <string>Edge Weight</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="doubleSpinBox_edge_weight">
                <property name="minimumSize">
                 <size>
                  <width>32</width>
                  <height>32</height>
                 </size>
                </property>
                <property name="minimum">
                 <double>-9999.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>9999.989999999999782</double>
                </property>
                <property name="singleStep">
                 <double>0.100000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="page_empty"/>
      </widget>
     </item>
     <item row="0" column="0" colspan="3">
      <widget class="QComboBox" name="comboBox_basic_adds">
       <property name="minimumSize">
        <size>
         <width>107</width>
         <height>32</height>
        </size>
       </property>
       <item>
        <property name="text">
         <string>Node</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Link</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1024</width>
     <height>36</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>32</width>
     <height>16</height>
    </size>
   </property>
   <property name="contextMenuPolicy">
    <enum>Qt::NoContextMenu</enum>
   </property>
   <property name="defaultUp">
    <bool>false</bool>
   </property>
   <property name="nativeMenuBar">
    <bool>true</bool>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="geometry">
     <rect>
      <x>1244</x>
      <y>580</y>
      <width>281</width>
      <height>190</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="minimumSize">
     <size>
      <width>32</width>
      <height>32</height>
     </size>
    </property>
    <property name="tearOffEnabled">
     <bool>false</bool>
    </property>
    <property name="title">
     <string>&amp;File</string>
    </property>
    <property name="icon">
     <iconset resource="content.qrc">
      <normaloff>:/Content/file_icon_white.png</normaloff>:/Content/file_icon_white.png</iconset>
    </property>
    <property name="separatorsCollapsible">
     <bool>false</bool>
    </property>
    <widget class="QMenu" name="menuOpen_Premises_File">
     <property name="title">
      <string>Open Premises File</string>
     </property>
     <addaction name="actionGEXF"/>
    </widget>
    <addaction name="actionNew"/>
    <addaction name="actionSave_Premises"/>
    <addaction name="menuOpen_Premises_File"/>
    <addaction name="actionEdit_User_Interface"/>
    <addaction name="actionSet_Virtual_Concierge_Background"/>
   </widget>
   <widget class="QMenu" name="menuBuild">
    <property name="minimumSize">
     <size>
      <width>0</width>
      <height>32</height>
     </size>
    </property>
    <property name="title">
     <string>Build</string>
    </property>
    <property name="icon">
     <iconset resource="content.qrc">
      <normaloff>:/Content/build_icon_white.png</normaloff>:/Content/build_icon_white.png</iconset>
    </property>
    <addaction name="actionRun"/>
   </widget>
   <widget class="QMenu" name="menuConfiguration">
    <property name="title">
     <string>Configuration</string>
    </property>
    <property name="icon">
     <iconset resource="content.qrc">
      <normaloff>:/Content/config_icon_white.png</normaloff>:/Content/config_icon_white.png</iconset>
    </property>
    <addaction name="actionInvert_Mouse_Y"/>
    <addaction name="actionDirectory_Wizard"/>
    <addaction name="actionEdit_Virtual_Concierge_Config"/>
    <addaction name="actionGrid"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menuBuild"/>
   <addaction name="menuConfiguration"/>
  </widget>
  <widget class="QDockWidget" name="dockWidget_configuration">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="minimumSize">
    <size>
     <width>320</width>
     <height>175</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>320</width>
     <height>175</height>
    </size>
   </property>
   <property name="contextMenuPolicy">
    <enum>Qt::DefaultContextMenu</enum>
   </property>
   <property name="features">
    <set>QDockWidget::AllDockWidgetFeatures</set>
   </property>
   <property name="windowTitle">
    <string>Configuration</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_configuration">
    <layout class="QGridLayout" name="gridLayout_10">
     <item row="0" column="0">
      <layout class="QGridLayout" name="gridLayout_9">
       <item row="0" column="0">
        <layout class="QGridLayout" name="gridLayout_7">
         <item row="0" column="0">
          <widget class="QLabel" name="label_floor_level">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Layer</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="spinBox_floor_level">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>32</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="1" column="0">
        <widget class="QCheckBox" name="checkBox_inversemouse_y">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>18</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>200</width>
           <height>13</height>
          </size>
         </property>
         <property name="text">
          <string>Invert Mouse Y</string>
         </property>
         <property name="icon">
          <iconset resource="content.qrc">
           <normaloff>:/Content/invert_icon_white.png</normaloff>:/Content/invert_icon_white.png</iconset>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QDockWidget" name="dockWidget_debug">
   <property name="minimumSize">
    <size>
     <width>87</width>
     <height>109</height>
    </size>
   </property>
   <property name="accessibleName">
    <string>Debugging Window</string>
   </property>
   <property name="windowTitle">
    <string>Debugging Window</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QGridLayout" name="gridLayout_8">
     <item row="0" column="0">
      <widget class="QTextBrowser" name="textBrowser">
       <property name="html">
        <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans Serif'; font-size:12px; font-weight:600; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'MS Shell Dlg 2'; font-size:12px;&quot;&gt;----Debug----&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionOpen_nodes_file">
   <property name="text">
    <string>Open Nodes File</string>
   </property>
  </action>
  <action name="actionRun">
   <property name="icon">
    <iconset resource="content.qrc">
     <normaloff>:/Content/run_icon_white.png</normaloff>:/Content/run_icon_white.png</iconset>
   </property>
   <property name="text">
    <string>Run</string>
   </property>
  </action>
  <action name="actionInvert_Mouse_Y">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="content.qrc">
     <normaloff>:/Content/invert_icon_white.png</normaloff>:/Content/invert_icon_white.png</iconset>
   </property>
   <property name="text">
    <string>Invert Mouse Y</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionEdit_User_Interface">
   <property name="icon">
    <iconset resource="content.qrc">
     <normaloff>:/Content/edit_icon_white.png</normaloff>:/Content/edit_icon_white.png</iconset>
   </property>
   <property name="text">
    <string>Edit User Interface</string>
   </property>
  </action>
  <action name="actionSync">
   <property name="text">
    <string>Sync</string>
   </property>
  </action>
  <action name="actionDirectory_Wizard">
   <property name="icon">
    <iconset resource="content.qrc">
     <normaloff>:/Content/wizard_white.png</normaloff>:/Content/wizard_white.png</iconset>
   </property>
   <property name="text">
    <string>Directory Wizard</string>
   </property>
  </action>
  <action name="actionEdit_Virtual_Concierge_Config">
   <property name="icon">
    <iconset resource="content.qrc">
     <normaloff>:/Content/config_icon_white.png</normaloff>:/Content/config_icon_white.png</iconset>
   </property>
   <property name="text">
    <string>Edit Virtual Concierge Config</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="icon">
    <iconset resource="content.qrc">
     <normaloff>:/Content/new_white_image.png</normaloff>:/Content/new_white_image.png</iconset>
   </property>
   <property name="text">
    <string>New</string>
   </property>
  </action>
  <action name="actionSave_Premises">
   <property name="icon">
    <iconset resource="content.qrc">
     <normaloff>:/Content/save_icon_white.png</normaloff>:/Content/save_icon_white.png</iconset>
   </property>
   <property name="text">
    <string>Save Premises</string>
   </property>
  </action>
  <action name="actionSet_Virtual_Concierge_Background">
   <property name="icon">
    <iconset resource="content.qrc">
     <normaloff>:/Content/background_icon_white.png</normaloff>:/Content/background_icon_white.png</iconset>
   </property>
   <property name="text">
    <string>Set Virtual Concierge Background</string>
   </property>
  </action>
  <action name="actionGEXF">
   <property name="text">
    <string>GEXF</string>
   </property>
  </action>
  <action name="actionGrid">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Snap to Grid</string>
   </property>
  </action>
  <zorder>dockWidget_configuration</zorder>
  <zorder>dockWidget_debug</zorder>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>RenderState</class>
   <extends>QOpenGLWidget</extends>
   <header>RenderStates/renderstate.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="content.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionInvert_Mouse_Y</sender>
   <signal>triggered(bool)</signal>
   <receiver>checkBox_inversemouse_y</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>129</x>
     <y>541</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
